src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.RenameFileEditor.cs

```diff
@@ -16,9 +16,21 @@ public RenameFileEditor(TService service, State state, string fileName, Cancella
             {
             }
 
-            internal override Task<ImmutableArray<CodeActionOperation>> GetOperationsAsync()
+            public override Task<ImmutableArray<CodeActionOperation>> GetOperationsAsync()
                 => Task.FromResult(RenameFileToMatchTypeName());
 
+            public override Task<Solution> GetModifiedSolutionAsync()
+            {
+                var oldDocument = SemanticDocument.Document;
+                var newDocumentId = DocumentId.CreateNewId(oldDocument.Project.Id, FileName);
+
+                var modifiedSolution = oldDocument.Project.Solution
+                    .RemoveDocument(oldDocument.Id)
```

Comment:
is there a reason we do this way rather than just change filename? do we not allow changing filename?

Reply:
Change file name is a code action that does exactly this. David also has a PR out that will change this a bit #33733